lib/block/blockdownload.js

```diff
@@ -112,40 +89,127 @@ class BlockDownload extends EventEmitter {
      * @property {Number} [retryCount] Retry count (default is 5)
      */
 
+    /**
+     * Create a a block download controller, which emits events
+     * on file (download) start, file (download) progress, file (download) end
+     * and file (download) error.
+     * @returns {TransferController} Transfer controller
+     */
+    createBlockDownloadController() {
+        const controller = new TransferController();
+        this.errorEvents = [];
+
+        controller.on(TransferEvents.CREATE_TRANSFER_PARTS, transferEvent => {
+            this.emit("filestart", transferEvent.transferAsset.eventData);
+        });
+
+        controller.on(TransferEvents.JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileprogress", {
+                ...transferEvent.transferAsset.eventData,
+                transferred: transferEvent.props.transferBytes
+            });
+        });
+
+        controller.on(TransferEvents.AFTER_JOIN_TRANSFER_PARTS, transferEvent => {
+            this.emit("fileend", transferEvent.transferAsset.eventData);
+        });
+
+        controller.on(TransferEvents.ERROR, transferEvent => {
+            console.log(`Error during block download: ${transferEvent.error}`);
+            this.errorEvents.push(transferEvent);
+
+            if (transferEvent.props.firstError) {
+                this.emit("fileerror", {
+                    ...transferEvent.transferAsset.eventData,
+                    errors: [DownloadError.fromError(transferEvent.error)]
+                });
+            }
+        });
+
+        console.log("Block download: block download controller created");
```

Comment:
shouldn't this be a debug log? this won't be very helpful for debugging if something went wrong with download since it is just setting up the controller. this should only run into issues in development. wdyt?

Reply:
I think it will be easier to piece together what happened if you have logs that allow to see object/variable lifetimes.